tests/role_wrapper.yml

```diff
@@ -0,0 +1,65 @@
+
+- name: Create a temporary file for a static inventory
+  tempfile:
+    state: file
+    suffix: .inventory.yaml
+  register: tempinventory
+  delegate_to: localhost
+
+- name: Create a static inventory from hostvars
+  template:
+    src: inventory.yaml.j2
+    dest: "{{ tempinventory.path }}"
+  delegate_to: localhost
+
+- name: Import role {{ tests_role_path }}
+  import_role:
+    name: "{{ tests_role_path }}"
+
+- name: >
+    Import role {{ tests_role_path }} via fresh Ansible instance with check
+    mode enabled
+  command: >
+    ansible all -vv -i {{ tempinventory.path }} --check -m import_role
+      -a "name={{ tests_role_path }}"
+  delegate_to: localhost
+  when:
+    - tests_test_checkmode is defined
+    - tests_test_checkmode
```

Comment:
these two lines can be combined into `tests_test_checkmode | d(false)` - same with the other similar `when` clauses below